components/CostsCard.vue

```diff
@@ -0,0 +1,290 @@
+            </p>
+          </div>
+        </div>
+        <div id="usdContainer" ref="usdContainer">
+          <div id="currency">
+            <p id="usdSymbol">
+              $
+            </p>
+            <p id="usdWord">
+              USD
+            </p>
+          </div>
+          <p id="totalCostPara" class="d-inline-block mb-0">
+            <span id="usdTotalCost">
+              <span>{{ totalCostAbbr?.amount }}</span>
+              <span id="totalCostUnit">
+                {{ totalCostAbbr?.unit }}
+              </span>
+            </span>
+          </p>
+          <p class="mt-0">
+            [insert note about GDP basis]
+          </p>
+        </div>
+      </div>
+      <CostsPie
+        v-if="pieStyle.height"
+        :hide-tooltips="hideTooltips"
+        :pie-size="styleWithLargestPie.height"
+        :style="pieStyle"
+        @mouseleave="onMouseLeavePie"
+        @mouseover="() => { hideTooltips = false }"
+      />
+    </div>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { abbreviateMillionsDollars } from "@/utils/money";
+import { CIcon } from "@coreui/icons-vue";
+
+const appStore = useAppStore();
+
+const totalCostAbbr = computed(() => {
+  if (appStore.totalCost) {
+    return abbreviateMillionsDollars(appStore.totalCost?.value, 1, true);
+  } else {
+    return undefined;
+  }
+});
+
+const hideTooltips = ref(false);
+const onMouseLeavePie = () => {
+  setTimeout(() => {
+    hideTooltips.value = true;
+  }, 500);
+};
+
+const cardBody = ref(null);
+const totalsContainer = ref(null);
+const gdpContainer = ref(null);
+const usdContainer = ref(null);
+const containers = {
+  cardBody: { ref: cardBody, width: ref(0), height: ref(0) },
+  totals: { ref: totalsContainer, width: ref(0), height: ref(0) },
+  gdp: { ref: gdpContainer, width: ref(0), height: ref(0) },
+  usd: { ref: usdContainer, width: ref(0), height: ref(0) },
+};
+
+// The first of three candidate styles that will be evaluated for how much space they can provide for the pie.
+const topRightStyle = computed(() => {
+  // What is the maximum size the pie can be if it is in the top right corner?
+  const availableWidthInTopRightCorner = containers.cardBody.width.value - Math.max(containers.gdp.width.value, containers.usd.width.value);
+  const availableHeightInTopRightCorner = containers.cardBody.height.value;
+  const maxSizeInTopRight = Math.min(availableWidthInTopRightCorner, availableHeightInTopRightCorner);
+  let rightPosition = 0;
+  if (maxSizeInTopRight < availableWidthInTopRightCorner) {
+    // If spare horizontal space, calculate a value for the 'right' CSS property for centering.
+    rightPosition = (availableWidthInTopRightCorner - maxSizeInTopRight) / 2;
+  }
+  return {
+    height: maxSizeInTopRight,
+    right: rightPosition,
+    top: 0,
+  };
+});
+
+// The second of three candidate styles that will be evaluated for how much space they can provide for the pie.
+const belowUsdTotalStyle = computed(() => {
```

Comment:
It might be a bit less confusing to call this `belowTotals` rather than `belowUsdTotal` - as it's slotting the pie entirely under the totals, rather than sneaking it into the corner between the two totals as bottomRight does. it's kind of irrelevant for this layout which total is lowest, as you calculate from `containers.total`